### ISSUE TITLE
onkeydown:stopPropagation breaks keyboard navigation

### ISSUE THREAD

**Issue by Cosmatevs** [radzenhq/radzen-blazor#2463]
**Describe the bug**
The commit https://github.com/radzenhq/radzen-blazor/commit/347942668d9003788403ec4e315ae2bfcec68263 introduced keyboard navigation, as well as `onkeydown:stopPropagation` in many components, such as Numeric.
As a result, if you have a component with keyboard navigation, and it contains a component that stops propagation of the `onkeydown` event, the keyboard navigation of the container component doesn't work if the stop-propagation component is active.

**To reproduce**
1. Go to the [ColorPicker](https://blazor.radzen.com/colorpicker) demo.
2. Open the color picker popup.
3. Navigate to one of the input fields, such as the HEX one.
4. Push the escape key.
5. Nothing happens.

**Expected behavior**
The escape key should close the dialog.

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Edge
 - Version 145.0.3800.82

**Additional context**
I've also seen `onkeypress:stopPropagation` in some components.

**Comment by enchev** [comment]
The case for ESC is indeed valid and should be fixed. Imagine howerver Numeric component inside the DataGrid cell and you press arrow up/down: in this case the Numeric should not propagate to the DataGrid since this will trigger row navigation.
